backend/capellacollab/settings/modelsources/t4c/models.py

```diff
@@ -79,6 +79,8 @@ class DatabaseT4CInstance(database.Base):
         back_populates="instance", cascade="all, delete"
```

Comment:
A server side validation would be good for all operations which are not permitted anymore. 

As far as I can see this only has an impact on linking T4C repositories to Capella models. We should prevent this for archives repositories.

Reply:
I've introduced a `get_existing_unarchived_instance` injectable and implemented its use within the `create_t4c_model` route, replacing the previous utilization of the `get_existing_instance` injectable. Ideally, the new injectable could have been constructed to directly utilize `get_existing_instance`. However, since we do not inject it but rather invoke it directly within the `create_t4c_model` route, adopting such an approach would introduce additional overhead.